cpp/README.md

```diff
@@ -279,7 +279,83 @@ both of these options would be used rarely. Current known uses-cases when they a
 
 *  Parameterized tests in google test.
 
+## Nightly Builds of `arrow-cpp`, `parquet-cpp`, and `pyarrow` for Linux
```

Comment:
Need to think about the best place to put this documentation. Maybe in the Python Sphinx area

Reply:
Yeah, I kept it here because it's related to all major arrow components not just Python

Reply:
We're running into the issue that we need to evolve the main Arrow site into cross-project documentation. Maybe we can do that when the site becomes a little better organized

Reply:
We could try to use https://breathe.readthedocs.io/en/latest/ to merge the C++ and Python documentation. As they are both very tightly coupled, I think they should have a common documentation. For example, the installation instructions are nearly the same.

Reply:
I'd be up for that. We could create a top-level doc directory. We also need to start writing Java documentation that is not Javadoc